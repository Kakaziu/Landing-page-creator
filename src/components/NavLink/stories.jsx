import NavLink from '.';
import links from './mock';

export default {
  title: 'NavLink',
  component: NavLink,
  args: {
    links: links,
  },
  argType: {
    links: { type: '' },
  },
};

export const Template = (args) => (
  <div>
    <NavLink {...args} />
  </div>
);
