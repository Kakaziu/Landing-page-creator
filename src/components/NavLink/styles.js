import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    color: ${theme.colors.primaryColor};
    display: flex;
    flex-flow: row wrap;

    @media ${theme.media.lteMedium} {
      flex-flow: column nowrap;
      align-items: center;
      align-content: center;
    }
  `}
`;
