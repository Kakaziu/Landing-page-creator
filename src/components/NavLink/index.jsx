import P from 'prop-types';
import * as Styled from './styles';
import MenuLink from '../MenuLink';

const NavLink = ({ links = [] }) => {
  return (
    <Styled.Container>
      {links.map((link) => (
        <MenuLink {...link} key={link.children} />
      ))}
    </Styled.Container>
  );
};

NavLink.propTypes = {
  links: P.arrayOf(
    P.shape({
      children: P.string.isRequired,
      link: P.string.isRequired,
      newTab: P.bool,
    }),
  ),
};

export default NavLink;
