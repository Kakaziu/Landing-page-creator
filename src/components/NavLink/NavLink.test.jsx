import renderTheme from '../../styles/render-theme';
import NavLink from '.';
import links from './mock';
import { screen } from '@testing-library/react';

describe('<NavLink/>', () => {
  it('should render all links', () => {
    renderTheme(<NavLink links={links} />);

    expect(screen.getAllByRole('link')).toHaveLength(links.length);
  });

  it('should not render links', () => {
    renderTheme(<NavLink />);

    expect(screen.queryAllByRole('link')).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLink links={links} />);

    expect(container).toMatchSnapshot();
  });
});
