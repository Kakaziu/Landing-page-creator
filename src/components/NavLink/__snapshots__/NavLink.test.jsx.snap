// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink/> should match snapshot 1`] = `
.c0 {
  color: #0A1128;
  display: flex;
  flex-flow: row wrap;
}

.c1 {
  color: #0A1128;
  font-size: 1.6rem;
  text-decoration: none;
  display: block;
  position: relative;
  padding: 1.6rem;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.5rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {
  .c0 {
    flex-flow: column nowrap;
    align-items: center;
    align-content: center;
  }
}

<div>
  <nav
    class="c0"
  >
    <a
      class="c1"
      href="#target1"
      target="_self"
    >
      Link 1
    </a>
    <a
      class="c1"
      href="#target2"
      target="_self"
    >
      Link 2
    </a>
    <a
      class="c1"
      href="#target3"
      target="_self"
    >
      Link 3
    </a>
    <a
      class="c1"
      href="#target4"
      target="_self"
    >
      Link 4
    </a>
    <a
      class="c1"
      href="#target5"
      target="_self"
    >
      Link 5
    </a>
    <a
      class="c1"
      href="#target6"
      target="_self"
    >
      Link 6
    </a>
    <a
      class="c1"
      href="#target7"
      target="_self"
    >
      Link 7
    </a>
    <a
      class="c1"
      href="#target8"
      target="_self"
    >
      Link 8
    </a>
    <a
      class="c1"
      href="#target9"
      target="_self"
    >
      Link 9
    </a>
    <a
      class="c1"
      href="#target10"
      target="_self"
    >
      Link 10
    </a>
  </nav>
</div>
`;
